### PR TITLE
[WIP] Progress on review

### PR DESCRIPTION
**Pull request by jpaddison3**

### CODE REVIEW

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 1]
```diff
@@ -140,51 +140,57 @@ const FrontpageReviewPhase = ({classes}: {classes: ClassesType}) => {
...
+  const forumTitle = forumTitleSetting.get()
+
+  const nominationPhaseDateRange = <span>{nominationStartDate.format('MMM Do')} – {nominationEndDate.format('MMM Do')}</span>
+  const reviewPhaseDateRange = <span>{nominationEndDate.clone().add(1, 'day').format('MMM Do')} – {reviewEndDate.format('MMM Do')}</span>
+  const votingPhaseDateRange = <span>{reviewEndDate.clone().add(1, 'day').format('MMM Do')} – {voteEndDate.format('MMM Do')}</span>
+
   // EA will use LW text next year, so I've kept the forumType genericization
   const overviewToolip = isEAForum ?
-    <>
+    <div>
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:151]: Huh, I'm curious what the issue was?

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 2]
```diff
@@ -140,51 +140,57 @@ const FrontpageReviewPhase = ({classes}: {classes: ClassesType}) => {
...
 
   const nominationsTooltip = isEAForum ?
-    <>
+    <div>
       <div>Nominate posts for the EA Writing Mega Review</div>
       <ul>
         <li>Any post from before 2021 can be nominated</li>
         <li>Any user registered before the start of the review can nominate posts</li>
-        <li>A post requires two nominations to progress to the review phase</li>
+        <li>Posts with at least one vote proceed to the Review Phase.</li>
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:181]:
```suggestion
        <li>Posts with at least one positive vote proceed to the Review Phase.</li>
```

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 3]
```diff
@@ -140,51 +140,57 @@ const FrontpageReviewPhase = ({classes}: {classes: ClassesType}) => {
...
       </ul>
-      <div>If you've been actively reading {siteNameWithArticleSetting.get()} before now, but didn't register an account, <Link href="/contact">contact us</Link></div>
-    </> :
-    <>
-      <div>Nominate posts for the {REVIEW_YEAR} Review</div>
+      <div>If you've been actively reading {siteNameWithArticleSetting.get()} before now, but didn't register an account, <Link to="/contact">contact us</Link></div>
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:183]: This was my bad, why did I think you could put links in a tooltip?

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 4]
```diff
@@ -212,7 +218,9 @@ const FrontpageReviewPhase = ({classes}: {classes: ClassesType}) => {
   }
 
   // NB: There's nothing ensuring that unnominated20NN is a valid filter
-  const allEligiblePostsUrl = `/allPosts?timeframe=yearly&after=${REVIEW_YEAR}-01-01&before=${REVIEW_YEAR+1}-01-01&limit=100&sortedBy=top&filter=unnominated${REVIEW_YEAR}`
+  const allEligiblePostsUrl = 
+    isEAForum ? `/allPosts?timeframe=yearly&after=${REVIEW_YEAR}-01-01&before=${REVIEW_YEAR+1}-01-01&limit=100&sortedBy=top&filter=unnominated` 
+    : `/allPosts?timeframe=allTime&after=2020-01-01&before=2021-01-01&limit=100&sortedBy=top&filter=unnominated`
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:223]: I think yearly is actually useful. It shows you the same information, but puts the year at the top, which is helpful orientation for users who aren't used to the review process. But it's your link, your call.

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 5]
```diff
@@ -228,18 +236,12 @@ const FrontpageReviewPhase = ({classes}: {classes: ClassesType}) => {
...
-          </Link>
-        </LWTooltip>
-      </SectionTitle>
+      />
       <div className={classes.reviewTimeline}>
         <div className={classes.nominationBlock}>
           <Link to={"/nominations"}>
             <LWTooltip placement="bottom-start" title={nominationsTooltip} className={classNames(classes.progress, {[classes.activeProgress]: activeRange === "NOMINATIONS"})}>
-              <div className={classNames(classes.blockText, classes.blockLabel)}>Nominations</div>
+              <div className={classNames(classes.blockText, classes.blockLabel)}>Rough Voting</div>
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:244]: Good

File: packages/lesswrong/components/review/ReviewVotingPage.tsx [hunk 6]
```diff
@@ -20,15 +20,16 @@ import { forumTypeSetting } from '../../lib/instanceSettings';
...
 const userVotesAreQuadraticField: keyof DbUser = "reviewVotesQuadratic2020";
 const isEAForum = forumTypeSetting.get() === "EAForum"
+const isLWForum = forumTypeSetting.get() === "LessWrong"
 
 export const currentUserCanVote = (currentUser: UsersCurrent|null) => {
   if (!currentUser) return false
-  if (!isEAForum && new Date(currentUser.createdAt) < new Date(`${REVIEW_YEAR}-01-01`)) return false
-  if (isEAForum && new Date(currentUser.createdAt) < new Date(annualReviewStart.get())) return false
+  if (isLWForum && new Date(currentUser.createdAt) > new Date(`${REVIEW_YEAR}-01-01`)) return false
+  if (isEAForum && new Date(currentUser.createdAt) > new Date(annualReviewStart.get())) return false
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/ReviewVotingPage.tsx:32]: Wait, seriously? You gotta be kidding. Those comparisons have to be the wrong way around. I didn't test it but I was like (internal feeling of) 98% confident. I gotta try. Very apologies if I was wrong.

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 7]
```diff
@@ -1249,11 +1253,11 @@ Posts.addView("nominatablePostsByVote", (terms: PostsViewTerms, _, context: Reso
   }
 })
 
-// Nominations for the (≤)2020 review are determined by the number of positive votes
-Posts.addView("roughVoting2020", (terms: PostsViewTerms) => {
+// Nominations for the (≤)2020 review are determined by the number of votes
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/posts/views.ts:1256]: Why you remove positive? The denormalized field checks for positive-ness.

File: packages/lesswrong/components/review/ReviewVotingPage.tsx (hunk 6 above)
Comment (jpaddison3) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx:32]: Nope, I was right.

Comment (jpaddison3) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx:32]: Asdf looking at the wrong component.

Comment (jpaddison3) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx:32]: I think I had what the function did turned around in my head. :\

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 2 above)
Comment (Raemon) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx:183]: I'm not sure we should bother doing this, although I do see the appeal. By the time we get to the review phase we'll already know which posts were overall sorted low, and we'll already need to encourage people to basically ignore low-ranked posts unless they think they're a hidden gem.

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 4 above)
Comment (Raemon) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx:239]: The problem is that it also weirdly truncates personal blogposts and emphasizes shortform. I agree with the year being at the top is nice.

**Review by Raemon — APPROVED** [review]

**Review by jpaddison3 — COMMENTED** [review]
I fixed the type errors and lint. ~I'd be interested in discussing the Expanded Post View.~

File: packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx:1]: Maybe you mean to add comments here? Oh there is something that shows unread comments. Nice.

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 4 above)
Comment (jpaddison3) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx:239]: What if we set `displayShortform=false`?

I'll add the url query, but I'll let you decide about personal blogposts.

File: packages/lesswrong/components/review/ReviewVotingPage.tsx (hunk 6 above)
Comment (jpaddison3) [reply packages/lesswrong/components/review/ReviewVotingPage.tsx]: Yep, except my pride

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 2 above)
Comment (jpaddison3) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx:183]: It already does it tho, check out the denormalized field filter

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+      </ul>
+      <div> Before the vote starts, you can try out the vote process on posts nominated and reviewed in {REVIEW_YEAR-1}</div>
+    </>
+
+  const dateFraction = (fractionDate: moment.Moment, startDate: moment.Moment, endDate: moment.Moment) => {
+    if (fractionDate < startDate) return 0
+    return ((fractionDate.unix() - startDate.unix())/(endDate.unix() - startDate.unix())*100).toFixed(2)
+  }
+
+  const allEligiblePostsUrl =
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:237]: Dumb question: If we're calling this all 2020 posts, why filter for unnominated?

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 10]
```diff
@@ -92,6 +93,9 @@ export const filters: Record<string,any> = {
   "unnominated2019": {
     nominationCount2019: 0
   },
+  "unnominated": {
+    reviewVoteCount: 0
+  },
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/posts/views.ts:98]: In other reason not to filter allposts with this, I'm not confident there's an index supporting this. `TODO`

**Comment by jpaddison3** [comment]
Boy this sure does get annoying:
image
(The hover-over is preventing me from clicking the vote button)

**Comment by jpaddison3** [comment]
I can awkwardly scroll on the reviewVoting page when there aren't that many posts

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 9 above)
Comment (Raemon) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx:237]: It should maybe have a different name. Filter for unnominated is so people can focus on adding new posts.

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx (hunk 1 above)
Comment (Raemon) [reply packages/lesswrong/components/review/FrontpageReviewWidget.tsx]: I think I just twiddled that randomly while trying to fix something, which it didn't help with, but then didn't feel motivated to change it back.

File: packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Raemon) [reply packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx:1]: Part of the point here was just that ReviewVotingPage was huge and unwieldy, and I wanted to break it up into pieces that were easier to think about.

File: packages/lesswrong/lib/collections/posts/views.ts (hunk 10 above)
Comment (Raemon) [reply packages/lesswrong/lib/collections/posts/views.ts:99]: We can just fix that tho

**Review by jpaddison3 — COMMENTED** [review]
We're pulling essentially 24 hour coverage in terms of code pushed to this branch, which is pretty cool 🏎

**Comment by jpaddison3** [comment]
I should say @Raemon, I really like your design sense.

**Comment by jpaddison3** [comment]
I don't love the way the top bar isn't part of the paper?
image

File: packages/lesswrong/server/resolvers/reviewVoteResolvers.ts [hunk 11]
```diff
@@ -28,25 +29,29 @@ addGraphQLResolvers({
...
-            ...(quadraticChange && !(typeof newQuadraticScore==='number') && {$inc: {
-              quadraticScore: quadraticChange
-            }})
-          }
-        )
+        // TODO:(Review) this could potentially introduce a race condition where
+        // the user does two increments in a row and the second read happens
+        // before the first write, leading to the discarding of the first
+        // increment. We should consider adding an increment option to
+        // updateMutator
```
Comment (jpaddison3) [inline packages/lesswrong/server/resolvers/reviewVoteResolvers.ts:36]: This is notable, but we have until we turn on quadratic voting to fix it.

File: packages/lesswrong/components/review/FrontpageReviewWidget.tsx [hunk 12]
```diff
@@ -0,0 +1,344 @@
...
+      <div>Cast your final votes for the {REVIEW_NAME_IN_SITU}. (Opens {reviewEndDate.clone().add(1, 'day').format('MMM Do')})</div>
+      <ul>
+        <li>Look over nominated posts and vote on them</li>
+        <li>Any user registered before {nominationStartDate.format('MMM Do')} can vote in the review</li>
+      </ul>
+    </> :
+    <>
+      <div>Cast your final votes for the {REVIEW_YEAR} Review. (Opens {reviewEndDate.clone().add(1, 'day').format('MMM Do')})</div>
+      <ul>
+        <li>Look over {/* TODO: Raymond Arnold look here, sentence fragment */} </li>
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/FrontpageReviewWidget.tsx:225]: @Raemon

File: packages/lesswrong/components/review/ReviewPostComments.tsx [hunk 13]
```diff
@@ -30,9 +32,11 @@ const PostReviewsAndNominations = ({ terms, classes, title, post, singleLine }:
...
   
   const { Loading, CommentsList, SubSection, CommentWithReplies, LoadMore } = Components
-
-  if (!loading && results && !results.length) {
-    return null
+  
+  // TODO: This doesn't quite work yet. Not sure why - Ray
+  const markAsRead = () => {
+    recordPostView({post, extraEventProperties: {type: "markAsRead"}})
+    setMarkedVisitedAt(new Date())
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/ReviewPostComments.tsx:39]: My opinion is that we can fix this later, but we should get to work on it asap.

File: packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+function recordPostView(arg0: { post: PostsListWithVotes; extraEventProperties: { type: string; }; }) {
+  throw new Error('Function not implemented.');
+}
+
+
+function setMarkedVisitedAt(arg0: Date) {
+  throw new Error('Function not implemented.');
+}
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/ReviewVotingExpandedPost.tsx:91]: 👀

Seems fine to push to not-this-PR

File: packages/lesswrong/components/review/ReviewVotingPage.tsx [hunk 15]
```diff
@@ -271,10 +214,11 @@ const ReviewVotingPage = ({classes}: {
 
   const [useQuadratic, setUseQuadratic] = useState(currentUser ? currentUser[userVotesAreQuadraticField] : false)
   const [loading, setLoading] = useState(false)
-  const [expandedPost, setExpandedPost] = useState<any>(null)
-  const [showKarmaVotes, setShowKarmaVotes] = useState<any>(null)
+  const [expandedPost, setExpandedPost] = useState<PostsListWithVotes|null>(null)
+  const [showKarmaVotes] = useState<any>(true)
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/ReviewVotingPage.tsx:218]: (Later) We can just remove whatever uses this conditional.

File: packages/lesswrong/components/review/ReviewVotingPage2019.tsx [hunk 16]
```diff
@@ -0,0 +1,617 @@
+import React, { useState, useEffect, useCallback } from 'react';
```
Comment (jpaddison3) [inline packages/lesswrong/components/review/ReviewVotingPage2019.tsx:1]: This page is not working. I think maybe @Raemon you decided that was ok with you? But if not then I think this is the first thing I've seen that would hold up shipping.

File: packages/lesswrong/lib/collections/posts/views.ts [hunk 17]
```diff
@@ -1247,3 +1253,42 @@ Posts.addView("stickied", (terms: PostsViewTerms) => (
...
+    }
+  }
+})
+ensureIndex(Posts,
+  augmentForDefaultView({ _id: 1, userId: 1, isEvent:1, baseScore:1 }),
+  { name: "posts.nominatablePostsByVote", }
+);
+
+
+// Nominations for the (≤)2020 review are determined by the number of votes
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/posts/views.ts:1278]:
```suggestion
// Nominations for the (≤)2020 review are determined by the number of positive votes
```

File: packages/lesswrong/lib/collections/votes/views.ts [hunk 18]
```diff
@@ -29,3 +34,24 @@ Votes.addView("tagVotes", function () {
...
+
+Votes.addView("userPostVotes", function ({voteType, collectionName, after/* , before */}, _, context: ResolverContext) {
+  return {
+    selector: {
+      collectionName: collectionName,
+      userId: context.currentUser?._id,
+      voteType: voteType,
+      cancelled: false,
+      isUnvote: false,
+      // $and: [{votedAt: {$gte: moment(after).toDate()}}, {votedAt: {$lt: moment(before).toDate()}}],
```
Comment (jpaddison3) [inline packages/lesswrong/lib/collections/votes/views.ts:46]:
```suggestion
```

**Review by jpaddison3 — COMMENTED** [review]
I just reviewed the whole damn thing. There's only one thing that might hold up shipping, on LW's end — that `/reviewVoting/2019` is borked. If @Raemon's fine with that, then :shipit:. Speaking on behalf of the Forum, this is ready to go.

**Pull request merged**